language/bert/bert_QDL.py

```diff
@@ -101,12 +101,18 @@ def get_sut_id_round_robin(self):
             self.next_sut_id = (self.next_sut_id + 1) % self.num_nodes
         return res
 
-    def client_predict(self, query, id):
+    def client_predict_worker(self, query, query_id):
         """Serialize the query, send it to the SUT in round robin, and return the deserialized response."""
         url = '{}/predict/'.format(self.sut_server_addr[self.get_sut_id_round_robin()])
-        #print(query)
-        response = requests.post(url, json={'query': query, id: id})
-        return response.json()['result']
+        responses = []
+        response = requests.post(url, json={'query': query})
```

Comment:
Thank you Arjun for the update of the code. 
If I am not mistaken it looks like there could be now multiple POSTs to the server as the number of threads. 
As we discuss in the Network call this allows to increase performance and reach the level of performance for non network operation of the current test.

I think maybe another optimization level can use an asynchronous API with async/await patterns and then not block the thread for queuing multiple posts for each thread. this might be needed when implementing the additional rate optimizations we discussed yesterday (batching in server and not client). let's consider the improvement if we see network rate limit. I have not tried it and maybe there are other methods so I am not certain.